notify/src/inotify.rs

```diff
@@ -296,6 +296,28 @@ impl EventLoop {
                                 );
                                 remove_watch_by_event(&path, &self.watches, &mut remove_watches);
                             }
+                            if event.mask.contains(EventMask::DELETE_SELF) {
+                                evs.push(
+                                    Event::new(EventKind::Remove(match path.clone() {
```

Comment:
Can you refactor this so there is no hidden match in the value creation ? Not sure but rustfmt might also move this around a fair bit..

Reply:
I took the logic to check for Dir Type outside push method.